datalad/customremotes/ria_utils.py

```diff
@@ -0,0 +1,158 @@
+    base_path : Path
+      Base path of the store.
+    dsid : str
+      Dataset ID
+
+    Returns
+    -------
+    Path, Path, Path
+      The location of the bare dataset repository in the store,
+      the directory with archive files for the dataset, and the
+      annex object directory are return in that order.
+    """
+    if version == 1:
+        dsgit_dir = base_path / dsid[:3] / dsid[3:]
+        archive_dir = dsgit_dir / 'archives'
+        dsobj_dir = dsgit_dir / 'annex' / 'objects'
+        return dsgit_dir, archive_dir, dsobj_dir
+    else:
+        raise ValueError("Unknown layout version: {}".format(version))
+
+
+def verify_ria_url(url, cfg):
+    """Verify and decode ria url
+
+    Expects a ria-URL pointing to a RIA store, applies rewrites and tries to
+    decode potential host and base path for the store from it. Additionally
+    raises if `url` is considered invalid.
+
+    ria+ssh://somehost:/path/to/store
+    ria+file:///path/to/store
+
+    Parameters
+    ----------
+    url : str
+      URL to verify an decode.
+    cfg : dict-like
+      Configuration settings for rewrite_url()
+
+    Raises
+    ------
+    ValueError
+
+    Returns
+    -------
+    tuple
+      (host, base-path, rewritten url)
+    """
+    from datalad.config import rewrite_url
+    from datalad.support.network import URL
+
+    if not url:
+        raise ValueError("Got no URL")
+
+    url = rewrite_url(cfg, url)
+    url_ri = URL(url)
+    if not url_ri.scheme.startswith('ria+'):
+        raise ValueError("Missing ria+ prefix in final URL: %s" % url)
+    if url_ri.fragment:
+        raise ValueError(
+            "Unexpected fragment in RIA-store URL: %s" % url_ri.fragment)
+    protocol = url_ri.scheme[4:]
+    if protocol not in ['ssh', 'file']:
+        raise ValueError("Unsupported protocol: %s" % protocol)
+
+    return url_ri.hostname if protocol == 'ssh' else None, url_ri.path, url
+
+
+def create_store(io, base_path, version):
+
+    # At store level the only version we know as of now is 1.
+    if version not in known_versions_dst:
+        raise UnknownLayoutVersion("store layout version unknown: {}"
+                                   "".format(version))
+
+    error_logs = base_path / 'error_logs'
+    version_file = base_path / 'ria-layout-version'
+    # TODO: Check base path for being empty?
+    #       Requires proper IO (command abstraction class).
+    #       But: Likely unnecessary. For now, check for version conflict only.
+    if io.exists(version_file):
+        existing_version = io.read_file(version_file).split('|')[0].strip()
+        if existing_version != version.split('|')[0]:
+            # We have an already existing location with a conflicting version on
+            # record.
+            # Note, that a config flag after pipe symbol is fine.
+            raise ValueError("Conflicting version found at target: {}"
+                             "".format(existing_version))
```

Comment:
I think it would be as fine with
```suggestion
                             .format(existing_version))
```